### ISSUE TITLE
Adam/AdamW bias correction uses one global step counter, mis-scaling a late-arriving parameter's first update

### ISSUE THREAD

**Issue by vyncint** [vyncint/oxmera#88]
## Summary

`AdamCore` keeps **one global step counter** for the whole optimizer and uses it for bias correction. A parameter whose gradient first arrives on step *k > 1* therefore gets its first moment estimate corrected as if it already had *k* steps of history, and its first update is mis-scaled.

Two parameters receiving identical gradients get different update magnitudes purely because one started later. This affects any model with conditionally-active parameters: a frozen block that unfreezes, a multi-task head trained on alternating batches, a mixture-of-experts router, or anything relying on 0.5.0's new "skip parameters without a gradient" behaviour (#53) — which makes late-arriving gradients a *supported* pattern rather than an accident.

## Reproduction

```rust
// oxmera-nn = "0.5.0", oxmera-optim = "0.5.0", oxmera-tensor = "0.5.0"
use oxmera_nn::Param;
use oxmera_optim::{Adam, Optimizer};
use oxmera_tensor::Tensor;

fn leaf(v: &[f32]) -> Param { Param::new(Tensor::from_slice(v, [v.len()]).unwrap()) }
fn grad(p: &Param) { p.value().mul_scalar(1.0).unwrap().sum(&[]).unwrap().backward().unwrap(); }

fn main() {
    let early = leaf(&[1.0]);
    let late  = leaf(&[1.0]);
    let mut opt = Adam::new(vec![early.clone(), late.clone()], 0.1);

    grad(&early);            // step 1: only `early` has a gradient
    opt.step().unwrap(); opt.zero_grad();

    grad(&late);             // step 2: `late` gets its FIRST gradient, same magnitude
    opt.step().unwrap();

    let d_early = 1.0 - early.value().to_vec_f32().unwrap()[0];
    let d_late  = 1.0 - late.value().to_vec_f32().unwrap()[0];
    println!("early={d_early:.6} late={d_late:.6} ratio={:.3}", d_late / d_early);
}
```

## Observed

```
early=0.100000 late=0.074413 ratio=0.744
```

Identical first gradients, first updates differing by 26%.

## Expected

```
early=0.100000 late=0.100000 ratio=1.000
```

Each parameter's bias correction should use **its own** update count, as in PyTorch, where `state['step']` is per parameter. Track the step count per parameter slot alongside `m`/`v` and compute `bias_correction1/2` from it.

## Where

`crates/oxmera-optim/src/lib.rs` — `AdamCore { step, m, v }`: `self.step` is global and `bc1`/`bc2` are computed once per `step()` call for all parameters.
